packages/@react-aria/tag/src/useTag.ts

```diff
@@ -10,42 +10,42 @@
  * governing permissions and limitations under the License.
  */
 
-import {ButtonHTMLAttributes, HTMLAttributes, KeyboardEvent, ReactNode} from 'react';
-import {DOMProps, Removable} from '@react-types/shared';
+import {ButtonHTMLAttributes, HTMLAttributes, KeyboardEvent, useRef} from 'react';
 import {filterDOMProps, mergeProps, useId} from '@react-aria/utils';
 // @ts-ignore
 import intlMessages from '../intl/*.json';
+import {useGridCell, useGridRow} from '@react-aria/grid';
 import {useMessageFormatter} from '@react-aria/i18n';
 
 
-export interface AriaTagProps extends Removable<ReactNode, void>, DOMProps {
-  children?: ReactNode,
-  isDisabled?: boolean,
-  validationState?: 'invalid' | 'valid',
-  isSelected?: boolean,
-  role?: string
-}
-
 export interface TagAria {
   tagProps: HTMLAttributes<HTMLElement>,
   labelProps: HTMLAttributes<HTMLElement>,
   clearButtonProps: ButtonHTMLAttributes<HTMLButtonElement>
 }
 
-export function useTag(props: AriaTagProps): TagAria {
+export function useTag(props, state): TagAria {
   const {
     isDisabled,
-    validationState,
     isRemovable,
-    isSelected,
     onRemove,
     children,
-    role
+    item
   } = props;
   const formatMessage = useMessageFormatter(intlMessages);
   const removeString = formatMessage('remove');
   const tagId = useId();
   const buttonId = useId();
+  let tagRef = useRef();
+  let labelRef = useRef();
+
+  let {rowProps} = useGridRow({
+    node: item
+  }, state, tagRef);
+  let {gridCellProps} = useGridCell({
+    node: item.childNodes[0],
```

Comment:
looks like the safer way to do this that we've done previously is https://github.com/adobe/react-spectrum/blob/main/packages/@react-spectrum/card/src/CardView.tsx#L175

Reply:
+1, we've done the same in ListViewItem and the drag examples, safer to clone the node